sdk/storage_blobs/src/blob_client.rs

```diff
@@ -120,4 +189,18 @@ mod tests {
             "10"
         )
     }
+
+    #[tokio::test]
+    async fn test_blob_types() {
+        // Create a Blob Client
+        let my_blob_client = BlobClient::<Unset>::new(
+            "vincenttranstock".to_string(),
+            "throwaway".to_string(),
+            "acontainer108f32e8".to_string(),
+            "hello.txt".to_string(),
+        );
+
+        let block_blob_client = my_blob_client.as_block_blob();
+        block_blob_client.upload_block_blob()
```

Comment:
![image](https://github.com/wastore/azure-sdk-for-rust-pr/assets/101599632/17fa0ed5-b7ec-4653-9f77-50f358bb39de)